list_unsigned_rpms

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python3
+
+from subprocess import check_output, STDOUT, CalledProcessError
+import argparse
+import glob
+
+
+def find_unsigned_packages(target_dir, gpgkey):
+    packages = glob.glob(f"{target_dir}/*.rpm")
+
+    for package in packages:
+        cmd = [
+            'rpm',
+            '--query',
+            '--queryformat',
+            '%{SIGPGP:pgpsig}',
+            package
+        ]
+
+        output = check_output(cmd, universal_newlines=True, stderr=STDOUT)
+
+        if gpgkey.lower() not in output:
+            yield package
+
+
+def handle_args():
+    parser = argparse.ArgumentParser(description='Sign unsigned RPMs in local stage repository')
+    parser.add_argument(
+        'repository_path',
+        help='Path to repository to sign unsigned RPMs'
+    )
+    parser.add_argument(
+        'gpgkey',
+        help='Short form gpgkey for the version being generated'
+    )
+
+    args = parser.parse_args()
+
+    if len(args.gpgkey) != 8:
```

Comment:
On Fedora 38 I see:
```console
$ rpm --query --queryformat '%{SIGPGP:pgpsig}' fedora-logos
RSA/SHA256, Mon 20 Feb 2023 19:01:58 CET, Key ID 809a8d7ceb10b464
```

Note how the Key ID is 16 characters long.

Reply:
Aye, with your method for the query the full length key is now available. Will update.